ckb-lock-common/src/intersection.rs

```diff
@@ -0,0 +1,38 @@
+// don't add extra code in the file.
+// it will be included in test cases in native code.
+
+pub fn get_intersection(
```

Comment:
suggest to use bulit-in `core::ops::Range` and `max/min` fn to simplify code:
```rust
fn get_intersection(range1: Range<usize>, range2: Range<usize>) -> Option<Range<usize>> {
    let max_start = range1.start.max(range2.start);
    let min_end = range1.end.min(range2.end);

    if max_start <= min_end {
        Some(max_start..min_end)
    } else {
        None
    }
}
```
and returning range struct can be used to index the slice directly:
```rust
        if let Some(intersection) = get_intersection((chunk_offset..chunk_offset+chunk.len()), target){
            chunk[intersection].fill(0);
        }
```

Reply:
Using Range is a good idea. However, your implementation is not perfect. What is `i.start < chunk_offset`  or i.end < chunk_offset? And `i.end-chunk_offset > chunk.size`. 
All the scenarios above are included in get_intersection in current implementation.

Reply:
Resolved by using Range, keep current implementation.

Reply:
> Using Range is a good idea. However, your implementation is not perfect. What is `i.start < chunk_offset` or i.end < chunk_offset? And `i.end-chunk_offset > chunk.size`. All the scenarios above are included in get_intersection in current implementation.

it won't happen because of the returning result is different with your current implementation